src/frontend/magnify/src/i18n/TranslationProvider/TranslationsProvider.tsx

```diff
@@ -9,31 +12,88 @@ export interface TranslationProviderProps {
   /**
    * The current locale
    */
-  locale: string;
+  locale?: string;
   /**
    * The default locale
    */
   defaultLocale: string;
-  /**
-   * Messages for the current locale
-   *
-   * @see loadLocaleData to fetch messages for a given locale for this lib
-   * You may need to merge it with your own project messages
-   */
-  messages: Record<string, string> | Record<string, MessageFormatElement[]> | undefined;
+
+  initTranslation?: boolean;
+}
+
+async function getTranslation(
+  locale: string,
+): Promise<Maybe<Record<string, string> | Record<string, MessageFormatElement[]>>> {
+  let translatedMessages: Maybe<Record<string, string> | Record<string, MessageFormatElement[]>>;
+  try {
+    translatedMessages = await loadLocaleData(locale);
+  } catch (error) {
+    translatedMessages = {};
+  }
+  return translatedMessages;
+}
+
+export interface LocaleContextInterface {
+  currentLocale: string;
+  setCurrentLocale: (locale: string) => void;
 }
 
+const LocaleContext = React.createContext<Maybe<LocaleContextInterface>>(undefined);
+
 export default function TranslationProvider({
   children,
-  locale,
+  locale = MagnifyLocales.FR,
   defaultLocale,
-  messages,
+  initTranslation = true,
 }: TranslationProviderProps) {
+  const [currentLocale, setCurrentLocale] = useState(
+    localStorage.getItem(MAGNIFY_LOCALE_KEY) ?? locale,
+  );
+  const [translations, setTranslations] = useState<
+    Maybe<Record<string, string> | Record<string, MessageFormatElement[]>>
+  >({});
+
+  const localeContext: LocaleContextInterface = useMemo(
+    () => ({
+      currentLocale: currentLocale,
+      setCurrentLocale: (locale: string) => {
+        localStorage.setItem(MAGNIFY_LOCALE_KEY, locale);
+        setCurrentLocale(locale);
+      },
+    }),
+    [currentLocale],
+  );
+
+  useEffect(() => {
+    if (!initTranslation) {
+      return;
+    }
+    getTranslation(currentLocale).then((initTranslation) => {
+      setTranslations(initTranslation ?? {});
+    });
+  }, [currentLocale]);
+
   return (
-    <IntlProvider defaultLocale={defaultLocale} locale={locale} messages={messages}>
-      {children}
-    </IntlProvider>
+    <LocaleContext.Provider value={localeContext}>
+      <IntlProvider
+        defaultLocale={defaultLocale}
+        locale={currentLocale}
+        messages={initTranslation ? translations : {}}
+      >
+        {children}
+      </IntlProvider>
+    </LocaleContext.Provider>
   );
 }
 
 export const useTranslations = useIntl;
+
+export const useLocale = () => {
```

Comment:
You should remove the empty file `src/frontend/magnify/src/hooks/locales/useLocale.ts`

Reply:
Arf...